### PR TITLE
Fix # 1810 : Irrelevant names are showing when searching a user in the private messaging

### LINKED ISSUE #1810: Irrelevant names are showing when searching a user in the private messaging
**Describe the bug**
When the user's first name, last name, or nickname was used or mentioned in other profile fields, when you try to search them in the private messaging, there are irrelevant names showing up. And in most cases, the user you are trying to message is not being listed.

**To Reproduce (in demo site)** please see: https://cutt.ly/LhrqfeZ
1. Create another profile field like "my best friend".
2. Edit some users and set "Trent" in their "my best friend" profile field.
3. Now, go to private messaging and try to search for "Trent" as you recipient.
4. Notice that irrelevant names are showing in the list and you cannot find Trent anymore.

**Expected behavior**
The private messaging search must be only limited to first name, last name and nickname because it will show unrelated or irrelevant names. It will be difficult to find the user.

**Screencast**
https://cutt.ly/LhrqfeZ

**Screenshots**
IMG

**Support ticket links**
https://secure.helpscout.net/conversation/1339832494/109164

### PR DESCRIPTION
**Pull request by manishpatolia**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #1810.

### How to test the changes in this Pull Request:

**To Reproduce (in demo site)** please see: https://cutt.ly/LhrqfeZ
1. Create another profile field like "my best friend".
2. Edit some users and set "XXXX" in their "my best friend" profile field.
3. Now, go to private messaging and try to search for "XXX" as you recipient.

### Proof Screenshots or Video

https://www.loom.com/share/8862d078681f4313bd7289cfb290c2e0

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-core/bp-core-functions.php [hunk 1]
```diff
@@ -3125,6 +3125,9 @@ function bp_core_get_suggestions( $args ) {
 		return new WP_Error( 'missing_parameter' );
 	}
 
+	//Remove action for remove search against xprofile fields
+	remove_action( 'bp_user_query_uid_clauses', 'bp_xprofile_bp_user_query_search', 10, 2 );
+
 	$suggestions = new $class( $args );
```
Comment (chetansatasiya) [inline src/bp-core/bp-core-functions.php:3131]: @manishpatolia This is correct like to remove the search from the Xprofile Fields but we can't remove the search from the Xprofile `First, Last & Nickname` field. So I would suggest you create a new function that will allow searching in Xprofile `First, Last & Nickname` field.

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@manishpatolia See my comment

File: src/bp-core/bp-core-functions.php (hunk 1 above)
Comment (manishpatolia) [reply src/bp-core/bp-core-functions.php]: Done as per your suggestion...

File: src/bp-core/bp-core-functions.php [hunk 2]
```diff
@@ -5321,3 +5332,72 @@ function bp_core_xprofile_clear_all_user_progress_cache() {
...
+ * @return array
+ */
+function bp_xprofile_search_bp_user_query_search_first_last_nickname( $sql, BP_User_Query $query ) {
+	global $wpdb;
+	if ( empty( $query->query_vars['search_terms'] ) || empty( $sql['where']['search'] ) ) {
+		return $sql;
+	}
+	$bp = buddypress();
+	$search_terms_clean = bp_esc_like( wp_kses_normalize_entities( $query->query_vars['search_terms'] ) );
+	if ( $query->query_vars['search_wildcard'] === 'left' ) {
```
Comment (chetansatasiya) [inline src/bp-core/bp-core-functions.php:5352]:
```suggestion
	if ( 'left'  === $query->query_vars['search_wildcard'] ) {
```

File: src/bp-core/bp-core-functions.php [hunk 3]
```diff
@@ -5321,3 +5332,72 @@ function bp_core_xprofile_clear_all_user_progress_cache() {
...
+		$search_terms_nospace = '%' . $search_terms_clean;
+		$search_terms_space   = '%' . $search_terms_clean . ' %';
+	} elseif ( $query->query_vars['search_wildcard'] === 'right' ) {
```
Comment (chetansatasiya) [inline src/bp-core/bp-core-functions.php:5355]:
```suggestion
	} elseif ( 'right' === $query->query_vars['search_wildcard']  ) {
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@manishpatolia do the requested changes

File: src/bp-core/bp-core-functions.php (hunk 2 above)
Comment (manishpatolia) [reply src/bp-core/bp-core-functions.php]: Changed

File: src/bp-core/bp-core-functions.php (hunk 3 above)
Comment (manishpatolia) [reply src/bp-core/bp-core-functions.php]: Changed

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@manishpatolia fix all the spacing issues in your code.

File: src/bp-core/bp-core-functions.php [hunk 4]
```diff
@@ -5325,3 +5336,72 @@ function bp_core_xprofile_clear_all_user_progress_cache() {
 	);
 
 }
+
+/**
+ * When search_terms are passed to BP_User_Query, search against xprofile fields.
+ *
+ * @since BuddyPress 2.0.0
```
Comment (chetansatasiya) [inline src/bp-core/bp-core-functions.php:5343]:
```suggestion
 * @since BuddyBoss 1.6.2
```

**Review by KartikSuthar — APPROVED** [review]
@manishpatolia Approved! Can you please merged to dev and pass to QA.

**Review by chetansatasiya — APPROVED** [review]
@manishpatolia merge in dev and add in testing.

@mohit-buddyboss @sumitbuddyboss make sure to test with the `Profile Names` settings.

**Comment by sumitbuddyboss** [comment]
Hello @manishpatolia 

When we search with any name like "john" and then remove the name then all the user name display which start with "J", Please check the below bug video.

Issue video:
https://www.loom.com/share/a4e2c76b9c3d44219f0b0701dafc1a97

cc @chetansatasiya @KartikSuthar 

Thank you

**Comment by manishpatolia** [comment]
> Hello @manishpatolia
> 
> When we search with any name like "john" and then remove the name then all the user name display which start with "J", Please check the below bug video.
> 
> Issue video:
> https://www.loom.com/share/a4e2c76b9c3d44219f0b0701dafc1a97
> 
> cc @chetansatasiya @KartikSuthar
> 
> Thank you

@sumitbuddyboss 

The above issue solved and also merge into the dev branch

Proof:
https://www.loom.com/share/a3b0eb163f4b4190a2fae5d6893f1fc4

@KartikSuthar @chetansatasiya

**Comment by sumitbuddyboss** [comment]
Hello @manishpatolia 

fixed > Related issue is resolved now relevant names are showing in the private message, for the proof please check the below video.

video
https://www.loom.com/share/939d06b4b6884a5580b3ee4d746db359

Ready to release 🚀 
@sourabhmatolia @chetansatasiya 

Thank you

**Review by chetansatasiya — APPROVED** [review]

**Pull request merged**